tools/flowc/backends/java/fi2java_compile.flow

```diff
@@ -802,21 +802,17 @@ emitJavaExpression(ctx : JavaScopeContext, expcode : FiExp, rtype : string) -> s
 			result(vs, "double");
 		}
 		FiString(s, __): {
-			vs = stringFold(s, "\"", \acc,code ->
-				acc + if (code < 0x20) {
-					if (code == 0x09) "\\t"
-					else if (code == 0x0a) "\\n"
-					else if (code == 0x0d) "\\r"
-					else "\\u"+lpad(formatHex(code), "0", 4);
-				} else if (code >= 0x80) {
-					"\\u" + lpad(formatHex(code), "0", 4)
-				} else {
-					if (code == 0x22) "\\\""
-					else if (code == 0x5c) "\\\\"
-					else fromCharCode(code);
-				}
-			) + "\"";
-			result(vs, "String");
+			s1 = toJavaStringLiteral(s);
+			// The length of a string constant in a class file is limited to 2^16 bytes in UTF-8 encoding
+			if (strlen(s1) > 32000) {
+				// A long string literal - can't be used directly
+				// TODO: We use UTF-16, so to be sure about long strings is it enough 32000 as a safe border? ...
+				long_string_ind = length(^(ctx.func.gctx.longStrings));
+				(ctx.func.gctx).longStrings := concat(^(ctx.func.gctx.longStrings), [s]);
+				result("LongStrings.longString" + i2s(long_string_ind) + "()", "String");
```

Comment:
Can we create this long string only once? Maybe rework function-member into a field?